transports/noise/src/lib.rs

```diff
@@ -213,8 +239,23 @@ pub enum Error {
     InvalidPayload(#[from] DecodeError),
     #[error(transparent)]
     SigningError(#[from] libp2p_identity::SigningError),
+    #[error("Unknown WebTransport certhashes: {}", certhashes_to_string(.0))]
+    UnknownWebTransportCerthashes(HashSet<Multihash>),
 }
 
 #[derive(Debug, thiserror::Error)]
 #[error(transparent)]
 pub struct DecodeError(quick_protobuf::Error);
+
+fn certhashes_to_string(certhashes: &HashSet<Multihash>) -> String {
+    let mut s = String::new();
+
+    for hash in certhashes {
+        if !s.is_empty() {
+            s.push_str(", ");
+        }
+        s.push_str(&bs58::encode(hash.to_bytes()).into_string());
+    }
+
+    s
+}
```

Comment:
Is this the most useful format? I would think that a user would expect the `multiaddr` `/certhash` format, i.e. the same format they can see in the server's multiaddr. In case you agree, you could wrap the `Multihash` in a `multiaddr::Protocol::Certhash` here and then use its `Display` implementation to get a string. Would also remove the additional dependency on `bs58`, though as noted above, we import it anyways.

Reply:
Do we care about the display order in case of multiple certhashes? Because with `HashSet` or `BTreeSet` the order will not be the same as the user provided them.

Reply:
I don't think the display order is relevant, especially since the handshake itself does not care about the order (correct me if I am wrong).